admin/installer/_images.py

```diff
@@ -67,3 +120,269 @@ def _unserialize_packer_dict(serialized_packer_dict):
+
+    :ivar build_region: The AWS region to build images in.
+    :ivar publish_regions: The AWS regions to publish the build images to.
+    :ivar template: The prototype configuration to use as a base. One of
+        `docker` or `flocker`.
+    :ivar distribution: The operating system distribution to install.
+        ubuntu-14.04 is the only one implemented so far.
+    :ivar configuration_directory: The directory containing prototype
+        configuration templates.
+    :ivar source_ami: The AMI ID to use as the base image.
+    :ivar working_directory: The directory in which templates will be copied
+        and modified.
+    """
+    build_region = field(type=unicode, mandatory=True)
+    publish_regions = pvector_field(item_type=unicode)
+    template = field(type=unicode, mandatory=True)
+    distribution = field(type=unicode, mandatory=True)
+    configuration_directory = field(type=FilePath, initial=PACKER_TEMPLATE_DIR)
+    source_ami = field(type=unicode, mandatory=True)
+    working_directory = field(type=FilePath, mandatory=True)
+
+
+@sync_performer
+def perform_packer_configure(dispatcher, intent):
+    """
+    Copy the prototype configuration files and provisioning scripts to a
+    temporary location and modify one of the configurations witht the values
+    found in ``intent``.
+    """
+    temporary_configuration_directory = intent.working_directory.child(
+        'packer_configuration'
+    )
+    temporary_configuration_directory.makedirs()
+    intent.configuration_directory.copyTo(temporary_configuration_directory)
+
+    template_name = (
+        u"template_{distribution}_{template}.json".format(
+            distribution=intent.distribution,
+            template=intent.template,
+        )
+    )
+    template_path = temporary_configuration_directory.child(
+        template_name
+    )
+
+    with template_path.open('r') as infile:
+        configuration = json.load(infile)
+
+    # If a region was specified, just build the image there and don't do
+    # any copying.
+    configuration['builders'][0]['region'] = intent.build_region
+    configuration['builders'][0]['source_ami'] = intent.source_ami
+    configuration['builders'][0]['ami_regions'] = thaw(intent.publish_regions)
+    output_template_path = template_path.temporarySibling()
+    with output_template_path.open('w') as outfile:
+        json.dump(configuration, outfile)
+    # XXX temporarySibling sets alwaysCreate = True for some reason.
+    output_template_path.alwaysCreate = False
+    return output_template_path
+
+
+class PackerBuild(PClass):
+    """
+    The attributes necessary to run ``packer build``.
+
+    # XXX This attribute should be called configuration_path
+    :ivar template: The path to a packer build configuration file.
+    :ivar reactor: The Twisted reactor object used to run the ``packer build``
+        subprocess.
+    :ivar sys_module: A ``sys`` like object with ``stdout`` and ``stderr``
```

Comment:
:question: Interesting thing to think about:

I'm not sure which is better, but my inclination would actually be to put sys_module and reactor into the performer rather than have them come in from the intent.

```
class PackerBuildPerformer(PClass):
   reactor = field(...)
   sys_module = field(...)
  
   def operator __call__(self, dispatcher, intent):
       self.reactor.run(intent.template)....
```

So on and so forth.

I'm nowhere near familiar enough with `Effect` to even evaluate the plusses and minuses of both implementations.